src/modules/75-cf/components/FlagArchiving/__tests__/ArchiveFlagButtons.test.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { render, RenderResult, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { TestWrapper } from '@common/utils/testUtils'
+import ArchiveFlagButtons, { ArchiveFlagButtonsProps } from '../ArchiveFlagButtons'
+
+const onArchiveMock = jest.fn()
+const onCloseMock = jest.fn()
+
+const renderComponent = (props: Partial<ArchiveFlagButtonsProps> = {}): RenderResult => {
+  return render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+    >
+      <ArchiveFlagButtons idMatch={false} onArchive={onArchiveMock} onClose={onCloseMock} {...props} />
+    </TestWrapper>
+  )
+}
+
+describe('ArchiveFlagButtons', () => {
+  test('it should disable the Archive button if the user types in a mismatching flag Id', async () => {
+    renderComponent()
+
+    expect(screen.getByRole('button', { name: 'archive' })).toBeDisabled()
+
+    userEvent.click(screen.getByRole('button', { name: 'archive' }))
+
+    expect(onArchiveMock).not.toHaveBeenCalled()
+  })
+
+  test('it should call onArchive callback if user types in matching flag Id', async () => {
+    renderComponent({ idMatch: true })
+
+    expect(screen.getByRole('button', { name: 'archive' })).toBeEnabled()
+
+    userEvent.click(screen.getByRole('button', { name: 'archive' }))
+
+    await waitFor(() => expect(onArchiveMock).toHaveBeenCalled())
+  })
```

Comment:
As mentioned above. Try to avoid out-of-scope things you assert on. If you were to kill off the `onArchiveMock` above the tests, you can rewrite this test pretty easily.
```suggestion
  test('it should call onArchive callback if user types in matching flag Id', async () => {
    const onArchiveMock = jest.fn()
    renderComponent({ idMatch: true, onArchive: onArchiveMock })

    expect(screen.getByRole('button', { name: 'archive' })).toBeEnabled()

    userEvent.click(screen.getByRole('button', { name: 'archive' }))

    await waitFor(() => expect(onArchiveMock).toHaveBeenCalled())
  })
```